File: backend/src/main/java/com/team05/petmeeting/domain/user/refreshtoken/repository/RefreshTokenRepository.java
package com.team05.petmeeting.domain.user.refreshtoken.repository;

import com.team05.petmeeting.domain.user.entity.User;
import com.team05.petmeeting.domain.user.refreshtoken.entity.RefreshToken;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface RefreshTokenRepository extends JpaRepository<RefreshToken, Long> {

    void deleteByToken(UUID token);

    Optional<RefreshToken> findByToken(UUID token);

    void deleteAllByUser(User user);

}

File: backend/src/main/java/com/team05/petmeeting/domain/user/refreshtoken/repository/RefreshTokenRepository.kt
package com.team05.petmeeting.domain.user.refreshtoken.repository

import com.team05.petmeeting.domain.user.entity.User
import com.team05.petmeeting.domain.user.refreshtoken.entity.RefreshToken
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface RefreshTokenRepository : JpaRepository<RefreshToken, Long> {

    fun deleteByToken(token: UUID)

    fun findByToken(token: UUID): RefreshToken?

    fun deleteAllByUser(user: User)
}
